python/lsst/ts/m2gui/controltab/tab_overview.py

```diff
@@ -152,6 +234,32 @@ def _update_control_status(self):
         )
         self._label_local_mode.setText(text_local_mode)
 
+    def _set_signal_status(self, signal_status):
```

Comment:
This makes the code absurdly complex. I believe that you missed the point of how signals shall be used - those aren't parameters, those are attributes which are always available (without need to construct objects). You really don't need to pass signal as argument, and all that use except for special cases shall be forbidden (see Qt itself doesn't pass signal along). If you include signal_status in Model (see comment in model.py), you will be able to hook to signal simply calling:
```
      Model.signal_status.connect(self._callback_signal_status)
```
in TabOverview constructor. No need to pass signal parameters along. Please consider changing that in all connect cases.

Reply:
This will induce the circular dependency that I tried to avoid. In my code design, the `TabOverview` will hold the `Model` as an internal attribute. Therefore, the `Model` should not have the knowledge of `TabOverview`, and I intentionally to abstract a function here to say the `TabOverview` needs to connect to a `SignalStatus`, which does not need to come from the `Model`. Maybe I will put that `SignalStatus` from `Model` to another class instead in the future to simplify the code in `Model`.

Thanks!

Reply:
Sorry, I was wrong (long time doing PySide2..), you need model. I still would suggest to pass model only as needed and don't store it otherwise, but fine for now - can comment on that for next PR.